erc1155/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+                account: account,
+                id: *ele,
+                amount: amount,
+            };
+
+            arr.push(obj);
+        }
+
+        return arr;
+    }
+
+    fn mint(&mut self, from: &ActorId, id: &u128, amount: u128) {
+        // check owner
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+        let old_balance = self.get_balance(from, id);
+        self.set_balance(from, id, old_balance.saturating_add(amount));
+
+        // TransferSingle event
+    }
+
+    fn mint_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch");
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+            let old_balance = self.get_balance(from, ele);
+            self.set_balance(from, ele, old_balance.saturating_add(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn burn_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
+        // TODO
+        // check owner
+        if from == &ZERO_ID {
+            panic!("ERC1155: burn from the zero address");
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch")
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+
+            let from_balance = self.get_balance(from, ele);
+
+            if from_balance < amount {
+                panic!("ERC1155: burn amount exceeds balance")
+            }
+
+            self.set_balance(from, ele, from_balance.saturating_sub(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn set_approval_for_all(&mut self, owner: &ActorId, operator: &ActorId, approved: bool) {
+        if owner == operator {
+            panic!("ERC1155: setting approval status for self")
+        }
+
+        self.operator_approvals
+            .entry(*owner)
+            .or_default()
+            .insert(*operator, approved);
+
+        // ApprovalForAll event
+    }
+
+    fn is_approved_for_all(&mut self, account: &ActorId, operator: &ActorId) -> &bool {
+        self.get_approval(account, operator)
+    }
+
+    fn get_approval(&mut self, owner: &ActorId, operator: &ActorId) -> &bool {
+        if owner == operator {
+            panic!("ERC1155: setting approval status for self")
+        }
+
+        self.operator_approvals
+            .entry(*owner)
+            .or_default()
+            .get(operator)
+            .unwrap_or(&false)
+    }
+
+    fn safe_transfer_from(&mut self, from: &ActorId, to: &ActorId, id: &u128, amount: u128) {
+        if from == to {
+            panic!("ERC1155: caller is not owner nor approved")
+        }
+
+        if *self.get_approval(from, to) {
```

Comment:
It must fail if the the function `get_approval` returns false.
```suggestion
        if !*self.get_approval(from, to) {
```